packages/gatsby-plugin-theme-ui/gatsby-node.js

```diff
@@ -0,0 +1,58 @@
+exports.onPreInit = (__, options) => {
+  let { themePreset } = options
+
+  if (typeof themePreset === 'string') {
+    try {
+      options.themePreset = require(themePreset)
+    } catch {
+      reporter.error(
+        `It appears your theme dependency is not installed. Try running \`${generateInstallInstructions()} ${themeModule}\``
+      )
+    }
+  }
+}
+
+function generateInstallInstructions() {
+  const { getConfigStore } = require(`gatsby-core-utils`)
+
+  const packageMangerConfigKey = `cli.packageManager`
+  const PACKAGE_MANGER = getConfigStore().get(packageMangerConfigKey) || `yarn`
+
+  const installKeyWord = PACKAGE_MANGER === `yarn` ? 'add' : 'install'
+
+  return `${PACKAGE_MANGER} ${installKeyWord}`
```

Comment:
This smells like something that could be useful as a utility for other plugins

Reply:
Agreed. Though we might be able to remove this now because we've updated the gatsby core webpack error. It's not quite as specific as this instruction set, but it does recommend the user install the dependency that's missing. Do we think that's sufficient? Or do we want both?

Reply:
Ah yeah... I think relying on Gatsby's error messaging here rather than a custom one here might make more sense TBH

Reply:
That's fair. Will update.

Reply:
My only question is if we want this to blow up the site? We could conceivably attempt the require and put a warning message instead?